include/interface/key.h

```diff
@@ -0,0 +1,198 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cctype>
+#include <iterator>
+#include <utility>
+
+namespace Input
+{
+
+//we only include the scancode we need, many or them are ignored, but the value
+//here is correct, based on * The values in this enumeration are based on the
+// USB usage page standard:
+// https://www.usb.org/sites/default/files/documents/hut1_12v2.pdf
+enum ScanCode {
+    ScanCode_Invalid = 0,
```

Comment:
Constant uses upper case.
```suggestion
    SCANCODE_INVALID = 0,
```